src/EventSubscriber/EventInstanceCanonicalMetatag.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Drupal\kdb_cludo\EventSubscriber;
+
+use Drupal\dpl_event\ReoccurringDateFormatter;
+use Drupal\recurring_events\Entity\EventInstance;
+use Drupal\recurring_events\Entity\EventSeries;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ViewEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Set canonical URL of eventinstances to eventseries if relevant.
+ *
+ * As we use field inheritance for eventinstances, a lot of the content will
+ * look as duplicate for search engines.
+ * If there are sibling eventinstances, we'll set the series as the canonical
+ * "owner" of the content.
+ */
+class EventInstanceCanonicalMetatag implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    private ReoccurringDateFormatter $reoccurringDateFormatter,
+  ) {}
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      KernelEvents::VIEW => ['onView', 100],
+    ];
+  }
+
+  /**
+   * If instance siblings, alter the canonical URL.
+   */
+  public function onView(ViewEvent $event): void {
+    $request = $event->getRequest();
+    $routeName = $request->attributes->get('_route');
+    $eventInstance = $request->attributes->get('eventinstance');
+
+    if ($routeName !== 'entity.eventinstance.canonical' || !$eventInstance instanceof EventInstance) {
+      return;
+    }
+
+    $eventSeries = $eventInstance->getEventSeries();
+
+    if (!($eventSeries instanceof EventSeries)) {
+      return;
+    }
+
+    $upcomingIds = $this->reoccurringDateFormatter->getUpcomingEventIds($eventSeries);
+
+    // We have a logic in dpl_event/EventSubscriber/EventSeriesRedirect that
+    // redirects EventSeries with a single active instance to the instance.
+    // If that's the case, we do not want to set the canonical link to a page
+    // that redirects back.
+    if (count($upcomingIds) > 1) {
+      $this->setCanonicalMetatag($event, $eventSeries);
+    }
```

Comment:
Consider if we should set the metatag for all instances instead of this the upcoming ones.

Past instances may stille be published after they occur and they may contain duplicated content just as well as the future ones do.